File: 0718/src/golang/Customer.go
package main

import "fmt"

type Person struct {
	name string
	age  int
}

type Customer struct {
	*Person
	memberID int
}

func (c Customer) enter() {
	fmt.Println("회원번호: ", c.memberID, " ", c.name+" , ", c.age, "세 입장")
}

func NewPerson(name string, age int) *Person {
	return &Person{name: name, age: age}
}

func NewCustomer(name string, age, memberID int) *Customer {
	return &Customer{
		Person:   NewPerson(name, age), // Person 객체 생성하여 Embedding
		memberID: memberID,
	}
}

func main() {
	// c := new(Customer)
	// c.name = "박자바"
	// c.age = 25
	// c.memberID = 1111
	// c.enter()
	// d := new(Customer)
	// d.name = "송코딩"
	// d.age = 20
	// d.memberID = 2222
	// d.enter()
	c := NewCustomer("박자바", 25, 1111)
	c.enter()
	d := NewCustomer("송코딩", 20, 2222)
	d.enter()
}

File: 0718/src/CTEST/Customer.java
package CTEST;

public class Customer extends Person{

	int memberID;
	
	public Customer() {}
	public Customer(String name, int age,int memberID) {
		super.name=name;
		super.age=age;
//		this.name=name;
//		this.age=age;
		this.memberID=memberID;
	}
	void enter() {
		System.out.println("회원번호: "+ memberID+" "+name+" , "+age+"세 입장");
	}
	
}
//this 사용해도 되는거같은데,
//상위클래스로 올려준다는 구분하기 좋은거같고, 
//this는 쓸때마다 한줄씩 잡아먹는데, 코드도 짧아지는거같음
